### DISCUSSION TITLE
Bring back language detection?

### DISCUSSION THREAD

**Discussion by thomykay** [elk-zone/elk#1164] [Ideas]
Any plans to have auto language detection in elk, Mastodon has removed it for whatever reason a year ago. https://github.com/mastodon/mastodon/pull/17478

Posting in 2 languages regularly, almost never take the time to set the language manually.

**Comment by thomykay** [comment]
Ok, there are some related issues, although they are not specifically asking for auto detection. https://github.com/elk-zone/elk/issues/967

**Comment by osmaa** [comment]
I was thinking the same. The CLD, ie i18n.detectLanguage() does indeed make mispredictions, but if it was limited to the following, I think it could be a big help:

- only confident guesses
- among only the languages in the user's preferences
- or among languages she has set by hand previously
- preferring the language in the thread, if used in a reply
- clearly highlighting the language selection widget when the guess is made
- the detected possibilities listed at the top of the popup menu when choosing language manually
